ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java

```diff
@@ -704,6 +746,55 @@ protected void deleteBlob(final String blobIdStr, final String serviceId, Future
       routerCallback.onPollReady();
     }
 
+    /**
+     * Requests for a blob to be undeleted asynchronously and invokes the {@link Callback} when the request completes.
+     * @param blobIdStr The ID of the blob that needs to be undeleted in string form
+     * @param serviceId The service ID of the service undeleting the blob. This can be null if unknown.
+     * @param futureResult A future that would contain information about whether the undelete succeeded or not,
+     *                     eventually.
+     * @param callback The {@link Callback} which will be invoked on the completion of a request.
+     */
+    protected void undeleteBlob(final String blobIdStr, final String serviceId, FutureResult<Void> futureResult,
+        final Callback<Void> callback) {
+      // Can skip GET if we can determine this is not a metadata bllob
```

Comment:
???

Looks like the implementation is also for GET?

Reply:
well, I have to do a GET to fetch all the blob ids for this blob if it's a composite blob.